core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -803,6 +840,7 @@ public void success(TaskInfo value)
                         currentRequest = null;
                         sendPlan.set(value.isNeedsPlan());
                         currentRequestStartNanos = HttpRemoteTask.this.currentRequestStartNanos;
```

Comment:
note that this callback might be called some time after request is finished, which means that if outbound DF came in the meantime it will resend DFs.

Reply:
This is being handled by the change to avoid starting next update until `currentRequest != null`
The success callback handler sets it back to `null` after taking a lock and the also updates sentDynamicFiltersVersion in critical section.